receiver/awscontainerinsightreceiver/internal/k8sapiserver/k8sapiserver.go

```diff
@@ -279,6 +291,137 @@ func (k *K8sAPIServer) getReplicaSetMetrics(clusterName, timestampNs string) []p
 	return metrics
 }
 
+func (k *K8sAPIServer) getPodMetrics(clusterName, timestampNs string) []pmetric.Metrics {
+	var metrics []pmetric.Metrics
+	podsList := k.leaderElection.podClient.PodInfos()
+	podKeyToServiceNamesMap := k.leaderElection.epClient.PodKeyToServiceNames()
+
+	for _, podInfo := range podsList {
+		if podInfo.Phase == corev1.PodPending {
+			fields := map[string]interface{}{}
+
+			if k.includeEnhancedMetrics {
+				addPodStatusMetrics(fields, podInfo)
+				addPodConditionMetrics(fields, podInfo)
+			}
+
+			attributes := map[string]string{
+				ci.ClusterNameKey: clusterName,
+				ci.MetricType:     ci.TypePod,
+				ci.Timestamp:      timestampNs,
+				ci.PodNameKey:     podInfo.Name,
+				ci.K8sNamespace:   podInfo.Namespace,
+				ci.Version:        "0",
+			}
+
+			podKey := k8sutil.CreatePodKey(podInfo.Namespace, podInfo.Name)
+			if serviceList, ok := podKeyToServiceNamesMap[podKey]; ok {
+				if len(serviceList) > 0 {
+					attributes[ci.TypeService] = serviceList[0]
+				}
+			}
+
+			if k.nodeName != "" {
```

Comment:
The pending pods will be emitted on the leader node, we have had discussion internally to change this, let me know what do you think

If no node name is passed, the log event is posted to the ```undefined``` log stream

Also see - the fallback logic is to replace with undefined  https://github.com/amazon-contributing/opentelemetry-collector-contrib/blob/aws-cwa-dev/exporter/awsemfexporter/grouped_metric.go#L52 If this fails as well, the log stream is set to  otel-stream-{UUID} from here - https://github.com/amazon-contributing/opentelemetry-collector-contrib/blob/aws-cwa-dev/exporter/awsemfexporter/emf_exporter.go#L112

Reply:
Updated